ledger/acctupdates.go

```diff
@@ -2328,6 +2399,37 @@ func catchpointRoundToPath(rnd basics.Round) string {
 	return outStr
 }
 
+// This function remove a single catchpoint file from the disk. this function does not leave empty directories
+func (au *accountUpdates) removeSingleCatchpointFileFromDisk(fileToDelete string) (err error) {
+	absCatchpointFileName := filepath.Join(au.dbDirectory, fileToDelete)
+	err = os.Remove(absCatchpointFileName)
+	if err == nil || os.IsNotExist(err) {
+		// it's ok if the file doesn't exist.
+		err = nil
+	} else {
+		// we can't delete the file, abort -
+		return fmt.Errorf("unable to delete old catchpoint file '%s' : %v", absCatchpointFileName, err)
+	}
+	catchpointRootDir := filepath.Join(au.dbDirectory, catchpointDirName)
+	currentCatchpointDir := filepath.Dir(absCatchpointFileName)
+
+	for currentCatchpointDir != catchpointRootDir {
```

Comment:
I can't say that this iteration won't work.. but it's making me a bit uncomfortable to reconstruct the directory paths and then loop until we get to our root. I think that a more robust solution here would be to use `filepath.SplitList` and *construct* the known path from the file itself. That path would not include the root, so you don't need to worry about your stop condition.